### ISSUE TITLE
Expression in 'When' clause of vscode extension does not work with resourceExtname if the extension to be compared has 2 dots

### ISSUE THREAD

**Issue by bergmeister** [Microsoft/vscode#66461]
Version: 1.31.0-insider (user setup)
Commit: ccdd2997a66965c7104fe180207d9a710ea50fa5
Date: 2019-01-11T06:19:08.621Z
Electron: 3.0.13
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763

Steps to Reproduce:

I am trying to add a file tab menu in an extension that only appears if the file ends in `.tests.ps1` (which is for PowerShell Pester tests that have to follow this naming convention because I do not want it to show for a normal PowerShell `.ps1` file) by defining the following in the `package.json` of the extension:
```json
      "editor/title/context": [
        {
          "when": "resourceExtname  == '.tests.ps1'",
          "command": "PowerShell.RunPesterTests"
        }
```
However, this does not work, using just `.ps1` in the expression works, showing that having 2 dots is somehow a problem. Just using `tests.ps1` in the condition does not work either

Does this issue occur when all extensions are disabled?: N.A.

**Comment by RMacfarlane** [comment]
Ah, thanks! I didn't read far enough down. I expected it to be at the top, after

> For conditional expressions, you can use either equality == ("editorLangId == typescript") or inequality != ("resourceExtname != .js").

**Comment by bergmeister** [comment]
@RMacfarlane Thanks for the tip with patterns/Regex, that was not easy to find in the docs (I did read the link provided by @usernamehw  before btw). Some docs or examples on the Regex syntax would be great as well since every language implements a slightly different version of Regex. I tried to use your suggestion `"when": "resourceFilename =~ /.*tests\\.ps1$/"` but it did not work, I also tried `"when": "resourceFilename =~ /\\.tests\\.ps1$/"`, which did not work either. I suppose I have to play with this a bit more...
